app_dart/lib/src/request_handlers/luci_status.dart

```diff
@@ -96,17 +97,34 @@ class LuciStatusHandler extends RequestHandler<Body> {
         userData['repo_owner'] as String,
         userData['repo_name'] as String,
       );
-      await githubChecksService.updateCheckStatus(
-        buildPushMessage,
-        luciBuildService,
-        slug,
-      );
+      await _updateCheckStatus(buildPushMessage, slug);
     } else {
       log.error('This repo does not support checks API');
     }
     return Body.empty;
   }
 
+  /// Updates check status with retries.
+  ///
+  /// The `githubChecksService.updateCheckStatus` has been hitting `SocketException` occasionally.
+  /// Retry logic is to mitigate this issue: https://github.com/flutter/flutter/issues/74611.
+  Future<void> _updateCheckStatus(BuildPushMessage buildPushMessage, RepositorySlug slug) async {
+    const RetryOptions r = RetryOptions(
+      maxAttempts: 3,
+      delayFactor: Duration(seconds: 2),
+    );
+    await r.retry(
+      () async {
+        await githubChecksService.updateCheckStatus(
```

Comment:
Should githubChecksService.updateCheckStatus handle the retries? that way any code using githubChecksService.updateCheckStatus will get the benefit.

Reply:
Retry logic has been available in `githubChecksUtil.updateCheckRun` when hitting  `GitHubError`.
Here the `SocketException` happens when calling `githubChecksService.updateCheckStatus`, but not the sub function inside. 
`githubChecksUtil.getCheckRun` is another sub function called inside, and the retry is added there as well so that other codes can benefit as well.

Reply:
I'm not sure I'm following. Given that the retry block is wrapping a single method and that method is already handling retries, wouldn't be easier/more reusable to just add SocketException to the list of retried exceptions?

Reply:
It is not clear which step of `githubChecksUtil.updateCheckRun` throws the `SocketException`. From log, it just throws one line error
```
SocketException: OS Error: Connection timed out, errno = 110, address = api.github.com, port = 40232
```
I am concerned that simply appending `SocketException` to the retried exception may not catch the error. Logs in https://github.com/flutter/flutter/issues/78555 makes me feel the `SocketException` happens elsewhere, otherwise it should catch the stacktrace?

Reply:
updateCheckRun has only two statements and the full method body is wrapped up in a retry. All this additional logic is equivalent to adding Socket exception here: https://cs.opensource.google/flutter/cocoon/+/master:app_dart/lib/src/foundation/github_checks_util.dart;l=73?q=GithubChecksUtil&ss=flutter

Reply:
Sorry I meant `githubChecksService.updateCheckStatus` from luci_status...

Reply:
seems only `updateCheckStatus` and ``getCheckRun` are related to github api call, as you suggested, I appended the `SocketException` to both functions instead of retry the call from `luci_status.dart`.
We can monitor how it goes with this change first before implementing retries in the upper call.